docs/ResourceInterface.md

```diff
@@ -24,121 +26,158 @@ public function getId(): int
+
+Transformer classes should extend the
+[`Sowl\JsonApi\AbstractTransformer`](/src/AbstractTransformer.php) class.
+
+[User Transformer Example](./examples/UserTransformer.php)
+
+### relationships()
+Returns the resource relationships.
+
 ```php
-class UserTransformer extends AbstractTransformer
+public static function relationships(): RelationshipsCollection
 {
-    public function transform(User $user): array
-    {
-        return [
-            'id' => $user->getId(),
-            'name' => $user->getName(),
-            'email' => $user->getEmail(),
-        ];
-    }
+    return new RelationshipsCollection();
 }
 ```
 
-## relationships()
-Method must return the resource relationships definitions.
-We need this method so that we can automatically serve all the relationships endpoints.
+This method must return an instance of
+[`Sowl\JsonApi\Relationships\RelationshipsCollection`](src/Relationships/RelationshipsCollection.php).
 
-Method must return [Sowl\JsonApi\Relationships\RelationshipsCollection](../src/Relationships/RelationshipsCollection.php).
+Two types of relationships exist, `To-One` and `To-Many`.
+
+#### To-One Relationship
+Returns a single resource in the responses.
 
-Example of resource without relationships:
 ```php
+// User Entity
 public static function relationships(): RelationshipsCollection
 {
-    return new RelationshipsCollection();
+    return new RelationshipsCollection([
+        ToOneRelationship::create('country', Country::class)
+    ]);
 }
 ```
 
-There are 2 types of relationships `To-One` and `To-Many` relationships.
+The `ToOneRelationship::create()` method accepts 3 parameters:
+
+- `name` (required):
+
+Name of the relationship that will be used in the endpoint:
+
+i.e: `GET /user/1/{name}` or `GET /user/1/relationships/{name}`
+
+i.e: `GET /user/1/country` or `GET /user/1/relationships/country`
+
+- `class` (required):
 
-### To-One
-To define new "To-One" relationship we must add new [Sowl\JsonApi\Relationships\ToOneRelationship](../src/Relationships/ToOneRelationship.php).
+The entity (resource) class being the relation.
 
-To-One relationship returns single resource in the responses.
+- `property` (optional):
 
-2 required parameters must be provider and 1 optional.
+Allow to set a custom value for the `{name}` for custom endpoints.
+
+For instance on the `User` entity, if the relationship is set as `protected Country $userCountry`,
+by default the endpoint would be `GET /user/1/userCountry`.
+
+```php
+// User Entity
+public static function relationships(): RelationshipsCollection
+{
+    return new RelationshipsCollection([
+        ToOneRelationship::create('userCountry', Country::class)
+    ]);
+}
+```
 
-#### name
-First parameter must be the name of relationships that will be used for generating building endpoints.
-For example `GET /user/1/{name}` or `GET /user/1/relationships/{name}`
+Set the 3rd optional parameter `property` to `country` to have a custom endpoint `GET /user/1/country`.
```

Comment:
To get the custom endpoint `/user/1/country` then you need to set next:
```php
        ToOneRelationship::create('country', Country::class, 'userCountry')
```

So name will be "country" and the property "userCountry" as we have the `protected Country $userCountry`